inference-engine/src/legacy_api/src/net_pass.cpp

```diff
@@ -272,14 +272,56 @@ void CombineData(DataPtr& master, DataPtr& slave) {
     }
 }
 
+/**
+ * Preserve output data name and update output data map of the network
+ *
+ * @param in_data name to update
+ * @param out_data name to preserve
+ * @param net output data map to update with in_data
+ */
+template <typename NET>
+void SaveOutputDataName(InferenceEngine::DataPtr in_data, InferenceEngine::DataPtr out_data, NET &net) {
+    // TODO: update outputs of the network if out_data was output
+    if (out_data->getInputTo().empty()) {
+        auto data_name = out_data->getName();
+        in_data->setName(data_name);
+    }
+}
+
+/**
+ * void SaveOutputDataName(InferenceEngine::DataPtr in_data, InferenceEngine::DataPtr out_data, NET &net), where
+ * NET = ICNNNetwork
+ */
+void SaveOutputDataName(InferenceEngine::DataPtr in_data, InferenceEngine::DataPtr out_data, ICNNNetwork& net) {
+    InferenceEngine::OutputsDataMap outputs_data_map;
+    net.getOutputsInfo(outputs_data_map);
+    if (out_data->getInputTo().empty()) {
+        auto out_data_name = out_data->getName();
+        in_data->setName(out_data_name);
+        if (outputs_data_map.count(out_data_name)) {
+            auto parent_layer_ptr = in_data->getCreatorLayer().lock();
```

Comment:
Please use IE_ASSERT to check that `parent_layer_ptr` is not equal nullptr.

Reply:
Good catch! Fixed.